File: app/src/main/java/com/himansh/movielist/MovieInfo.java
package com.himansh.movielist;

import android.app.ProgressDialog;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;
import android.widget.Toast;

import com.android.volley.Request;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.ImageLoader;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.NetworkImageView;

import org.json.JSONException;
import org.json.JSONObject;

public class MovieInfo extends AppCompatActivity {

    TextView Title, Released, Rating, Genre, Director, Writer, Actor, Plot;
    NetworkImageView Poster;
    private ProgressDialog pDialog;
    private String url = "http://www.omdbapi.com/?i=tt0099785&apikey=94a221d";
    ImageLoader imageLoader = AppController.getInstance().getImageLoader();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_movie_info);

        Poster = findViewById(R.id.imageViewPoster);

        url = "http://www.omdbapi.com/?i=" + getIntent().getStringExtra("imdbID") + "&apikey=94a221d";

        Title = findViewById(R.id.textViewTitle);
        Released = findViewById(R.id.textViewReleased);
        Rating = findViewById(R.id.textViewIMDBRatings);
        Genre = findViewById(R.id.textViewGenre);
        Director = findViewById(R.id.textViewDirector);
        Writer = findViewById(R.id.textViewWriter);
        Actor = findViewById(R.id.textViewActors);
        Plot = findViewById(R.id.textViewPlot);

        pDialog = new ProgressDialog(this);
        pDialog.setMessage("Loading...");
        pDialog.show();

        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest
                (Request.Method.GET, url, null, new Response.Listener<JSONObject>() {

                    @Override
                    public void onResponse(JSONObject response) {
                        hidePDialog();
                        try {

                            Title.setText(response.getString("Title"));
                            Released.setText(response.getString("Released"));
                            Genre.setText("Genre: " + response.getString("Genre"));
                            Director.setText("Director: " + response.getString("Director"));
                            Writer.setText("Writer: " + response.getString("Writer"));
                            Actor.setText("Cast: " + response.getString("Actors"));
                            Plot.setText("Plot:\n" + response.getString("Plot"));
                            Rating.setText(response.getString("imdbRating") + "/10");

                            if (imageLoader == null)
                                imageLoader = AppController.getInstance().getImageLoader();
                            NetworkImageView thumbNail = (NetworkImageView) findViewById(R.id.imageViewPoster);

                            thumbNail.setImageUrl(response.getString("Poster"), imageLoader);


                        } catch (JSONException e) {
                            e.printStackTrace();
                        }

                    }
                }, new Response.ErrorListener() {

                    @Override
                    public void onErrorResponse(VolleyError error) {
                        showError(error);
                        hidePDialog();

                    }
                });


        AppController.getInstance().addToRequestQueue(jsonObjectRequest);
    }

    private void showError(VolleyError error) {
        Toast.makeText(this, "I'll handle this Error if you hire me!\n\n" + error.toString(), Toast.LENGTH_LONG).show();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        hidePDialog();
    }

    private void hidePDialog() {
        if (pDialog != null) {
            pDialog.dismiss();
            pDialog = null;
        }
    }
}

File: app/src/main/java/com/himansh/movielist/ui/MovieInfo.kt
package com.himansh.movielist.ui


import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.himansh.movielist.R
import com.himansh.movielist.data.model.MovieObject
import com.himansh.movielist.data.remote.RepoService
import com.himansh.movielist.data.remote.RetrofitClientInstance
import com.himansh.movielist.databinding.ActivityMovieInfoBinding
import com.himansh.movielist.util.ProgressDialog
import com.squareup.picasso.Picasso
import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

class MovieInfo : AppCompatActivity() {

    companion object {
        private const val API_KEY = "94a221d"
    }

    private var pDialog: ProgressDialog? = null
    private lateinit var repoService: RepoService
    private lateinit var binding: ActivityMovieInfoBinding

    private lateinit var movieID: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = DataBindingUtil.setContentView(this, R.layout.activity_movie_info)
        movieID = intent.getStringExtra("imdbID").toString()

        repoService = RetrofitClientInstance.retrofitInstance.create(RepoService::class.java)
        ProgressDialog.initialise(this)

        getMovieDetails()

    }

    private fun getMovieDetails() {

        ProgressDialog.show()

        val cryptoObservable: Observable<MovieObject> = repoService.getMovieDetail(movieID, API_KEY)
        cryptoObservable.subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(this::handleResults, this::handleError)
    }

    private fun handleResults(marketList: MovieObject?) {
        ProgressDialog.dismiss()
        binding.movieObject = marketList

        Picasso.get()
                .load(marketList?.Poster)
                .into(binding.moviePoster)
    }

    private fun handleError(t: Throwable) {
        Toast.makeText(this, "ERROR IN FETCHING API RESPONSE. Try again",
                Toast.LENGTH_LONG).show()
        showError(t.message!!)
        ProgressDialog.dismiss()
    }

    private fun showError(error: String) {
        Toast.makeText(this, "I'll handle this Error if you hire me!\n\n$error", Toast.LENGTH_LONG).show()
    }

    public override fun onDestroy() {
        super.onDestroy()
        hidePDialog()
    }

    private fun hidePDialog() {
        if (pDialog != null) {
            pDialog!!.dismiss()
            pDialog = null
        }
    }
}